src/Fixer/ConstantNotation/NativeConstantInvocationFixer.php

```diff
@@ -0,0 +1,211 @@
+use PhpCsFixer\FixerConfiguration\FixerOptionBuilder;
+use PhpCsFixer\FixerDefinition\CodeSample;
+use PhpCsFixer\FixerDefinition\FixerDefinition;
+use PhpCsFixer\Tokenizer\CT;
+use PhpCsFixer\Tokenizer\Token;
+use PhpCsFixer\Tokenizer\Tokens;
+use Symfony\Component\OptionsResolver\Exception\InvalidOptionsException;
+
+/**
+ * @author Filippo Tessarotto <zoeslam@gmail.com>
+ */
+final class NativeConstantInvocationFixer extends AbstractFixer implements ConfigurationDefinitionFixerInterface
+{
+    /**
+     * @var array<string, true>
+     */
+    private $constantsToEscape = [];
+
+    /**
+     * @var array<string, true>
+     */
+    private $caseInsensitiveConstantsToEscape = [];
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getDefinition()
+    {
+        return new FixerDefinition(
+            'Add leading `\` before constant invocation of internal constant to speed up resolving. Constant name match is case-sensitive, except for `null`, `false` and `true`.',
+            [
+                new CodeSample('<?php var_dump(PHP_VERSION, M_PI, MY_CUSTOM_PI);'.PHP_EOL),
+                new CodeSample(
+                    '<?php var_dump(PHP_VERSION, M_PI, MY_CUSTOM_PI);'.PHP_EOL,
+                    [
+                        'include' => [
+                            'MY_CUSTOM_PI',
+                        ],
+                    ]
+                ),
+                new CodeSample(
+                    '<?php var_dump(PHP_VERSION, M_PI, MY_CUSTOM_PI);'.PHP_EOL,
+                    [
+                        'fix_built_in' => false,
+                        'include' => [
+                            'MY_CUSTOM_PI',
+                        ],
+                    ]
+                ),
+                new CodeSample(
+                    '<?php var_dump(PHP_VERSION, M_PI, MY_CUSTOM_PI);'.PHP_EOL,
+                    [
+                        'exclude' => [
+                            'M_PI',
+                        ],
+                    ]
+                ),
+            ],
+            null,
+            'Risky when any of the constants are namespaced or overridden.'
+        );
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function isCandidate(Tokens $tokens)
+    {
+        return $tokens->isTokenKindFound(T_STRING);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function isRisky()
+    {
+        return true;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function configure(array $configuration = null)
+    {
+        parent::configure($configuration);
+
+        $uniqueConfiguredExclude = \array_unique($this->configuration['exclude']);
+
+        // Case sensitive constants handling
+        $constantsToEscape = array_values($this->configuration['include']);
+        if (true === $this->configuration['fix_built_in']) {
+            $constantsToEscape = array_merge($constantsToEscape, \array_keys(\get_defined_constants()));
```

Comment:
`get_defined_constants` also includes userland constants (which would be constants define by the run of the PHP-CS-Fixer in such case). This could cause issues (and contradicts the fixer name saying `native`)
A solution would be to use the categorized output of `get_defined_constants`, to remove constants belonging to the `user` category.

Reply:
`user` category now skipped